12-deploy-python-microservice.ipynb

```diff
@@ -0,0 +1,730 @@
+    " * Running on http://10.0.2.100:8080\n",
+    "```\n",
+    "\n",
+    "After that you can navigate in your browser on the same environment to http://127.0.0.1:8080/health and should see the `App is running` message."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "2136243d",
+   "metadata": {},
+   "source": [
+    "#### Creating a github repository.\n",
+    "\n",
+    "We want to create a github repository following this [link.](https://docs.github.com/en/repositories/creating-and-managing-repositories/quickstart-for-repositories)\n",
+    "\n",
+    "Once you have your repository, modify the above bash command to clone it."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e4e4c39",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "git clone YOUR-GIT-REPOSITORY ~/smart-village-service-repository\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "f1b272a9",
+   "metadata": {},
+   "source": [
+    "Now we are going to copy the `app.py` and `Containerfile` files to the git repository directory and enter the directory."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "c5d777e1",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "cp ~/smartvillage-service/* ~/smart-village-service-repository/\n",
+    "cd ~/smart-village-service-repository/\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "d1c14f9e",
+   "metadata": {},
+   "source": [
+    "Next step is to commit and push our python script and Containerfile to our online repository. \n",
+    "If you want update the README.md file for documentation purposes now is also a good moment to do that."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9cfadd79",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%%bash\n",
+    "git add app.py\n",
+    "git commit -m \"app: Initial python microservice code\"\n",
+    "git push origin\n",
+    "echo DONE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ca889e1e",
+   "metadata": {},
+   "source": [
+    "## Deploying on Openshift\n",
+    "Openshift has a cool technology called source to image([s2i](https://github.com/openshift/source-to-image)). The code we pushed to our github repository can be easily imported into Openshift and the system will create, a container build similar to the build we did locally. Push it to a registry and start a deployment using that image. It will also setup all the routes and services needed to talk to the service.\n",
+    "\n",
+    "However, remember we need to pass 3 environment variables to the container after the deployment we need to update it to include them.\n",
+    "\n",
+    "1. From your [developer sandbox](https://console.redhat.com/openshift/sandbox) open the openshift view.\n",
+    "2. In the Developer view, select Add+ and click on the Import from Git tile.\n",
```

Comment:
We might want to use the "Container images" tile instead of the "Import from Git" tile, and work with our image built in Quay.io instead.

Reply:
ohhh... I see. Sorry working my way down from here.